pkg/chunk/dynamodb_table_client.go

```diff
@@ -156,34 +156,51 @@ func (d dynamoTableClient) DescribeTable(ctx context.Context, name string) (desc
 	return
 }
 
-func (d dynamoTableClient) UpdateTable(ctx context.Context, desc TableDesc) error {
-	var tableARN *string
-	if err := d.backoffAndRetry(ctx, func(ctx context.Context) error {
-		return instrument.TimeRequestHistogram(ctx, "DynamoDB.UpdateTable", dynamoRequestDuration, func(ctx context.Context) error {
-			out, err := d.DynamoDB.UpdateTableWithContext(ctx, &dynamodb.UpdateTableInput{
-				TableName: aws.String(desc.Name),
-				ProvisionedThroughput: &dynamodb.ProvisionedThroughput{
-					ReadCapacityUnits:  aws.Int64(desc.ProvisionedRead),
-					WriteCapacityUnits: aws.Int64(desc.ProvisionedWrite),
-				},
-			})
-			if err != nil {
+func (d dynamoTableClient) UpdateTable(ctx context.Context, current, expected TableDesc) error {
+
+	if current.ProvisionedRead != expected.ProvisionedRead || current.ProvisionedWrite != expected.ProvisionedWrite {
+		if err := d.backoffAndRetry(ctx, func(ctx context.Context) error {
+			return instrument.TimeRequestHistogram(ctx, "DynamoDB.UpdateTable", dynamoRequestDuration, func(ctx context.Context) error {
+				_, err := d.DynamoDB.UpdateTableWithContext(ctx, &dynamodb.UpdateTableInput{
+					TableName: aws.String(expected.Name),
+					ProvisionedThroughput: &dynamodb.ProvisionedThroughput{
+						ReadCapacityUnits:  aws.Int64(expected.ProvisionedRead),
+						WriteCapacityUnits: aws.Int64(expected.ProvisionedWrite),
```

Comment:
That seems to have been working all along so far - providing both when only one has changed.